scripts/jenkins/ardana/ansible/roles/heat_stack/tasks/delete.yml

```diff
@@ -23,6 +23,20 @@
         cloud: "{{ os_cloud }}"
         name: "{{ heat_stack_name }}"
         state: absent
+
+  rescue:
+    - name: Force delete servers from stack '{{ heat_stack_name }}'
+      shell: |
+         openstack --os-cloud {{ os_cloud }} stack resource list --filter \
+           type=OS::Nova::Server -f value -c physical_resource_id {{ heat_stack_name }} |\
+           awk '{print "openstack --os-cloud '{{ os_cloud }}' server \
+           delete --wait "$1 }' | sh -x || :
+
+    - name: Retry delete stack '{{ heat_stack_name }}'
+      os_stack:
+        cloud: "{{ os_cloud }}"
+        name: "{{ heat_stack_name }}"
+        state: absent
       register: stack_status
       retries: 2
```

Comment:
with this second task in place you may run `openstack stack delete` up to 3 times, which is no longer aligned with the timeouts we use in the `openstack-ardana-heat` pipeline. Either remove the retry (preferable, because if you can't delete the stack two times in a row, you should give up anyway, based on our last observations), or increase the timeout.

Reply:
done